fili-core/src/test/groovy/com/yahoo/bard/webservice/table/PermissiveConcretePhysicalTableSpec.groovy

```diff
@@ -0,0 +1,85 @@
+import static org.joda.time.DateTimeZone.UTC
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.data.dimension.BardDimensionField
+import com.yahoo.bard.webservice.data.dimension.DimensionColumn
+import com.yahoo.bard.webservice.data.dimension.MapStoreManager
+import com.yahoo.bard.webservice.data.dimension.impl.KeyValueStoreDimension
+import com.yahoo.bard.webservice.data.dimension.impl.ScanSearchProviderManager
+import com.yahoo.bard.webservice.data.metric.MetricColumn
+import com.yahoo.bard.webservice.metadata.TestDataSourceMetadataService
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.Interval
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class PermissiveConcretePhysicalTableSpec extends Specification {
+    @Shared PermissiveConcretePhysicalTable permissiveConcretePhysicalTable
+
+    @Shared DimensionColumn disjointIntervalColumn
+    @Shared MetricColumn leftAbuttingIntervalColumn, rightAbuttingIntervalColumn
+
+    @Shared Interval leftAbuttingInterval
+    @Shared Interval rightAbuttingInterval
+    @Shared Interval disjointInterval
+
+    def setupSpec() {
+        disjointIntervalColumn = new DimensionColumn(
+                new KeyValueStoreDimension(
+                        "disjointIntervalColumn",
+                        null,
+                        [BardDimensionField.ID] as LinkedHashSet,
+                        MapStoreManager.getInstance("dimension"),
+                        ScanSearchProviderManager.getInstance("apiProduct")
+                )
+        )
+        leftAbuttingIntervalColumn = new MetricColumn("leftAbuttingIntervalColumn")
+        rightAbuttingIntervalColumn = new MetricColumn("rightAbuttingIntervalColumn")
+
+        disjointInterval = new Interval("2018-01-01/2018-02-01")
+        leftAbuttingInterval = new Interval("2017-01-01/2017-02-01")
+        rightAbuttingInterval = new Interval("2017-02-01/2017-03-01")
+
+        permissiveConcretePhysicalTable = new PermissiveConcretePhysicalTable(
+                TableName.of('test table'),
+                DAY.buildZonedTimeGrain(UTC),
+                [disjointIntervalColumn, leftAbuttingIntervalColumn, rightAbuttingIntervalColumn] as Set,
+                ['dimension': 'druidDim'],
+                new TestDataSourceMetadataService([
+                        (disjointIntervalColumn)     : [disjointInterval] as Set,
+                        (leftAbuttingIntervalColumn) : [leftAbuttingInterval] as Set,
+                        (rightAbuttingIntervalColumn): [rightAbuttingInterval] as Set,
+                        (new Column('ignored'))      : [new Interval('2010-01-01/2500-12-31')] as Set
+                ])
+        )
+    }
+
+    @Unroll
+    def "The spanning intervals are returned even when individual columns are not fully covering the interval"() {
```

Comment:
If we have a data-driven test (ie. a test with a `where` block), we should _always_ use `@Unroll`. The reason is that `@Unroll` makes a test (as seen by the JUnit Test Runner) for each "row" in a `where` table. Without that, all of the iterations in the `where` block are run as a single test, so a failure with the 1st row marks the entire test as a failure, instead of allowing _each_ iteration to fail on it's own, which gives much more information.